### PR TITLE
Replace plotnine implementation of barplot with seaborn/matplotlib

### PR DESCRIPTION
**Pull request by kcArtemis**
Removed the dependency (plotnine)

**PR Checklist**

-   [x] Referenced issue is linked
-   [ ] If you've fixed a bug or added code that should be tested, add tests!
-   [ ] Documentation in `docs` is updated

**Description of changes**

In this pull request, I addressed the issue #425 by removing the dependency on plotnine. Instead, I have implemented the use of matplotlib and seaborn for plotting. Specifically, I updated the `barplot` function and the `perturbscore` function to utilize these libraries, providing a more lightweight solution for the project.

**Technical details**

While making these changes, I ensured that the functionality remains consistent, and the transition to matplotlib and seaborn is seamless. The choice of these libraries was made to minimize dependencies and provide a more straightforward solution for users. No additional dependencies are introduced, and the overall performance is maintained.

**Additional context**

### CODE REVIEW

**Review by Zethson — COMMENTED** [review]
Great!
Thank you very much!

1. Could you please remove plotnine from the dependencies from `pyproject.toml`?
2. Could you please paste a before and after plot here? Like one with the old plotnine implementation and one with your new implementation
3. We have a lot of hardcoded plot parameters now (like `s`, `alpha,`, and more). Which ones do you think we should expose to the user? Should we populate them at least with kwargs?

Thanks!

File: pertpy/plot/_mixscape.py [hunk 1]
```diff
@@ -237,10 +233,9 @@ def perturbscore(  # pragma: no cover
         # If before_mixscape is True, split densities based on original target gene classification
         if before_mixscape is True:
             cols = {gd: "#7d7d7d", target_gene: color}
-            p = ggplot(perturbation_score, aes(x="pvec", color=labels)) + geom_density() + theme_classic()
-            p_copy = copy.deepcopy(p)
-            p_copy._build()
-            top_r = max(p_copy.layers[0].data["density"])
+            p = sns.kdeplot(data=perturbation_score, x="pvec", hue=labels, fill=False, common_norm=False)
```
[the remark this answers is not in the document]
Comment (kcArtemis) [reply pertpy/plot/_mixscape.py:235]: In the previous version: the name of the variable assigned  was p, can I change it??

**Comment by kcArtemis** [comment]
> Great! Thank you very much!
> 
> 1. Could you please remove plotnine from the dependencies from `pyproject.toml`?
> 2. Could you please paste a before and after plot here? Like one with the old plotnine implementation and one with your new implementation
> 3. We have a lot of hardcoded plot parameters now (like `s`, `alpha,`, and more). Which ones do you think we should expose to the user? Should we populate them at least with kwargs?
> 
> Thanks!

we can expose s and alpha to the user since s:marker size and alpha: transparency parameter, for that we can make the use of kwargs.

File: pertpy/plot/_mixscape.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Zethson) [reply pertpy/plot/_mixscape.py:235]: Yes!

**Comment by Zethson** [comment]
[images_pertpy_mixscale.pdf](https://github.com/theislab/pertpy/files/13415062/images_pertpy_mixscale.pdf)

Before and after by @kcArtemis

File: pertpy/plot/_mixscape.py [hunk 2]
```diff
@@ -309,41 +294,32 @@ def perturbscore(  # pragma: no cover
...
+                    data=perturbation_score, x="pvec", hue="mix", fill=True, common_norm=False, palette=cols, alpha=0.7
+                )
+                sns.scatterplot(
+                    data=perturbation_score, x="pvec", y="y_jitter", hue="mix", palette=cols, s=10, alpha=0.5
                 )
-        return p2
+                pl.xlabel("Perturbation score", fontsize=16)
+                pl.ylabel("Cell density", fontsize=16)
+                pl.title("Density Plot using Seaborn and Matplotlib", fontsize=18)
+                pl.legend(title="mix", title_fontsize=14, fontsize=12)
```
Comment (Zethson) [inline pertpy/plot/_mixscape.py:319]: The title `mix` is a bit off. Could we rename this to `mixscape class`, please?

**Comment by subwaystation** [comment]
:hotsprings:

**Pull request merged**